database.go

```diff
@@ -53,54 +52,31 @@ func isValidPassword(username string, password string) (int, bool) {
 	return 0, false
 }
 
-func getUserIDFromSessionHash(sessionHash string) (int, error) {
+func getUserIDFromSessionHash(sessionHash string) (int, int, error) {
 	// Then it checks if the session is in the SESSIONS table
-	stmt, err := db.Prepare("SELECT user_id FROM SESSIONS WHERE session_hash=?")
+	stmt, err := db.Prepare("SELECT SESSIONS.user_id AS id, usertype_id FROM SESSIONS, USERS WHERE session_hash=? AND USERS.user_id=id;")
 	if err != nil {
-		return -1, err
+		return -1, -1, err
 	}
 	defer stmt.Close()
 	rows, err := stmt.Query(sessionHash)
 	if err != nil {
-		return -1, err
+		return -1, -1, err
 	}
 	defer rows.Close()
 	var userID int
+	var usertypeID int
 	if rows.Next() {
-		err = rows.Scan(&userID)
+		err = rows.Scan(&userID, &usertypeID)
 		if err != nil {
-			return -1, err
+			return -1, -1, err
 		}
-		return userID, nil
+		return usertypeID, userID, nil
```

Comment:
I think here is a problem about the number's ID, this function return first userTypeID but in server.go, line 56, the call to getUserIDFromSessionHash save this first return on userUD

Reply:
Good catch. Fixed